src/main/resources/types/Result.js

```diff
@@ -78,6 +81,156 @@ angular.module('guacConntest').factory('Result', [function defineResult() {
+        // connection, taking inaccuracy into account
+        var logarithmicRTT = Math.log((stats.median + stats.medianAbsoluteDeviation) / Result.WORST_TOLERABLE_LATENCY + 1) / Math.LN2;
+
+        // Map logarithmically-scaled RTT onto integer bins, where 0 is the
+        // subjectively best possible connection and higher values are
+        // subjectively worse
+        return Math.min(Result.NICENESS_BINS - 1, Math.floor(logarithmicRTT * (Result.NICENESS_BINS - 1)));
+
+    };
+
+    /**
+     * Compresses the given set of connection test results, producing an opaque
+     * string which may be passed to Result.unpack() to reconstitute those
+     * results. Redundant information may be stripped from the results for the
+     * sake of saving space. The given results are assumed to be from a
+     * completed connection test.
+     *
+     * @param {Result[]} results
+     *     The set of connection test results to compress.
+     *
+     * @returns {String}
+     *     An opaque string which, if passed to Result.unpack(), will produce
+     *     the given set of connection test results.
+     */
+    Result.pack = function pack(results) {
+
+        // Strip calculated values from results, including only server URLs and
+        // the samples measured for each server
+        var samplesByUrl = {};
+        angular.forEach(results, function addResult(result) {
+            if (result.roundTripStatistics)
+                samplesByUrl[result.server.url] = result.roundTripStatistics.samples;
+        });
+
+        // Convert packed results to JSON, compress using deflate, and finally
+        // encode that binary value as a base64 string
+        return btoa(pako.deflateRaw(angular.toJson(samplesByUrl), { 'to' : 'string' }));
+
+    };
+
+    /**
+     * Decompresses the given opaque string producing the original set of
+     * connection test results. The given string MUST have come from a previous
+     * call to Result.pack(). If the provided set of servers differs from that
+     * available at the time Result.pack() was called, results for servers no
+     * longer present will be omitted.
+     *
+     * @param {Object.<String, Server>} servers
+     *     A map of @link{Server} objects, where the key of each entry is the
+     *     human-readable name for the Guacamole server described by the
+     *     corresponding @link{Server} object. This map should be the current
+     *     set of servers as would be returned through a call to
+     *     configService.getServers().
+     *
+     * @param {String} packed
+     *     An opaque string generated by a previous call to Result.pack().
+     *
+     * @returns {Result[]}
+     *     The set of connection test results originally provided to
+     *     Result.pack() when the opaque string was generated.
+     */
+    Result.unpack = function unpack(servers, packed) {
+
+        // Decode the given base64 string into (presumably) deflated JSON data,
```

Comment:
What do you mean by "presumably"?

Reply:
If the base64 comes from a wiley user that made up the data rather than a previous call to `Result.pack()`, then the deflate operation will produce gibberish which will fail to parse as JSON, ultimately resulting in an exception.

Reply:
Ok, that makes sense.